code/modules/clothing/modular_armor/attachments/modules.dm

```diff
@@ -204,6 +204,142 @@
 	chemsystem.RemoveComponent()
 	return ..()
 
+
+/obj/item/armor_module/module/eshield
+	name = "Arrowhead Energy Shield System"
+	desc = "A brand new innovation in armor systems, this device creates a shield around the user that is capable of negating all damage. If it sustains too much it will deactivate, and, due to the install requirements, leave the user completely vunerable."
+	icon = 'icons/mob/modular/modular_armor_modules.dmi'
+	icon_state = "mod_eshield"
+	item_state = "mod_eshield_a"
+	slot = ATTACHMENT_SLOT_MODULE
+	variants_by_parent_type = list(/obj/item/clothing/suit/modular/xenonauten = "mod_eshield_xn", /obj/item/clothing/suit/modular/xenonauten/light = "mod_eshield_xn", /obj/item/clothing/suit/modular/xenonauten/heavy = "mod_eshield_xn")
+
+	///Current shield Health
+	var/shield_health = 0
+	///Maximum shield Health
+	var/max_shield_health = 40
+	///Amount to recharge per tick, processes once every two seconds.
+	var/recharge_rate = 8
+
+	///Spark system used to generate sparks when the armor takes damage
+	var/datum/effect_system/spark_spread/spark_system
+
+	///Shield color when the shield is 0 - 33% full
+	var/shield_color_low = COLOR_MAROON
+	///Shield color when the shield is 33 - 66% full
+	var/shield_color_mid = COLOR_MOSTLY_PURE_RED
+	///Shield color when the shield is 66% to full
+	var/shield_color_full = COLOR_BLUE_LIGHT
+
```

Comment:
these dont need to be vars

Reply:
david wants to add different versions later on, this is so we can have different colors depending on shield type